R/nth.R

```diff
@@ -0,0 +1,38 @@
+nth_default <- function(x, default = NA) {
+  vec_assert(default, size = 1)
+  vec_cast(default, x)
+}
+
+#' @export
+nth <- function(x, n, default = NA) {
+  x_size <- vec_size(x)
+
+  if (x_size == 0) {
+    return(nth_default(x, default))
+  }
+
+  if (!is_integerish(n)) {
```

Comment:
```suggestion
  if (!is_integerish(n, n = 1L, finite = TRUE)) {
```

I think we need the `n = 1L` check, but also consider `finite = TRUE` to prevent `nth(x, Inf)`?

Reply:
yes thanks